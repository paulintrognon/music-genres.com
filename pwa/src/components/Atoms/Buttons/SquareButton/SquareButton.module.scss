.container {
  display: inline-flex;
  background-color: var(--color-tertiary);
  justify-content: center;
  align-items: center;
  box-shadow: 0 9px 10px 0 rgba(0,0,0,.23);
  padding: 1rem 3rem;
  transition: background-color .2s,border .2s,color .2s;
  text-transform: uppercase;
  font-family: Montserrat;
  font-weight: 600;
  font-size: 1.1rem;
  width: auto;

  img {
    margin-right: 0.5rem;
  }
  
  &:hover {
    background-color: var(--color-tertiary-light);
  }
}
