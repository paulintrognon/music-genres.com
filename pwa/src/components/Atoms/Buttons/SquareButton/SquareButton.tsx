import classnames from 'classnames'
import Loader from '../../Loader/Loader'
import classes from './SquareButton.module.scss'

interface Props extends React.ComponentProps<'span'> {
  isLoading?: boolean
}
const SquareButton: React.FC<Props> = ({ isLoading, className, children, ...props }) => {
  return (
    <span className={classnames(classes.container, className)} {...props}>
      {isLoading ? <Loader /> : children}
    </span>
  )
}
export default SquareButton
