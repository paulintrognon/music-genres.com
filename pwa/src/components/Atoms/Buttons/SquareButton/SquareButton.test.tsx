import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import SquareButton from './SquareButton'

const renderer = createRenderer()

it('should render the SquareButton', () => {
  expect(renderer.render(<SquareButton>Content</SquareButton>)).toMatchSnapshot()
})

it('should render with a loader if loading', () => {
  expect(renderer.render(<SquareButton isLoading>Content</SquareButton>)).toMatchSnapshot()
})
