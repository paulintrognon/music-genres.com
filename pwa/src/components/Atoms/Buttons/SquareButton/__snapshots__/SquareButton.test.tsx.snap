// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the SquareButton 1`] = `
<span
  className="container"
>
  Content
</span>
`;

exports[`should render with a loader if loading 1`] = `
<span
  className="container"
>
  <Loader />
</span>
`;
